torch/csrc/distributed/rpc/utils.h

```diff
@@ -142,50 +127,37 @@ struct TORCH_API LazyStreamContext {
 
   // get a stream for the given device. If it is the first time using that
   // device, allocate a new stream and store it in the map.
-  c10::Stream getStream(c10::DeviceIndex index) {
-    if (!stream_creator_) {
-      throw std::runtime_error(c10::str(
-          "Attempting to access device stream of device ",
-          index,
-          ", but the device doesn't support streams"));
-    }
-    auto iter = streams_.find(index);
+  c10::Stream getStream(c10::Device device) {
+    auto iter = streams_.find(device);
     if (iter == streams_.end()) {
-      auto stream = stream_creator_(index);
-      streams_.emplace(index, stream);
+      auto stream = impl_->getStreamFromPool(device);
+      streams_.emplace(device, stream);
       return stream;
     } else {
       return iter->second;
     }
   }
 
-  std::set<c10::DeviceIndex> devices() const {
-    std::set<c10::DeviceIndex> devices;
+  std::unordered_set<c10::Device> devices() const {
```

Comment:
Is this method being used anywhere? There didn't seem to be any consequences to changing the return type...

Reply:
it's used here:

https://github.com/pytorch/pytorch/blob/65968ab817db323a532f50a2f2ea131ae27dada5/torch/csrc/distributed/rpc/tensorpipe_agent.cpp#L758